File: backend/app/schemas/proposal.py
from pydantic import BaseModel


class ProposalRow(BaseModel):
    proposal_type: str
    target_count: int
    target_amount: float
    actual_count: int
    actual_amount: float


class ProposalCrossRow(BaseModel):
    region: str
    proposals: dict[str, int]


class ProposalDetail(BaseModel):
    customer_name: str | None
    region: str | None
    deal_content: str | None
    new_deal_amount: float
    received_amount: float
    record_date: str | None

File: frontend/src/api/proposal.ts
import client, { type ApiResponse } from './client'

export interface ProposalRow {
  proposal_type: string
  sub_proposal_type: string | null
  target_count: number
  actual_count: number
  target_amount: number
  actual_amount: number
}

export interface ProposalCrossRow {
  region: string
  proposals: Record<string, number>
}

export const fetchProposalOverview = () =>
  client.get<ApiResponse<ProposalRow[]>>('/v1/proposal/overview').then(r => r.data.data)

export const fetchProposalCrossTable = () =>
  client.get<ApiResponse<ProposalCrossRow[]>>('/v1/proposal/cross-table').then(r => r.data.data)

export interface ProposalDetailRow {
  customer_name: string | null
  region: string | null
  deal_content: string | null
  new_deal_amount: number
  received_amount: number
  record_date: string | null
}

export const fetchProposalDetail = (region?: string, proposal_type?: string) =>
  client.get<ApiResponse<ProposalDetailRow[]>>('/v1/proposal/detail', { params: { region, proposal_type } }).then(r => r.data.data)
